helical/models/geneformer/fine_tuning_model.py

```diff
@@ -87,26 +87,41 @@ def __init__(self,
 
         self.fine_tuning_head.set_dim_size(self.config["embsize"])
 
-    def _forward(self, input_ids: torch.Tensor, attention_mask_minibatch: torch.Tensor) -> torch.Tensor:
+    def _forward(self, input_ids: torch.tensor, attention_mask_minibatch: torch.tensor, original_lengths: torch.tensor) -> torch.tensor:
         """
         Forward method of the fine-tuning model.
 
         Parameters
         ----------
-        input_ids : torch.Tensor
+        input_ids : torch.tensor
             The input ids to the fine-tuning model.
-        attention_mask_minibatch : torch.Tensor
+        attention_mask_minibatch : torch.tensor
             The attention mask for the input tensor.
+        original_lengths: torch.tensor
+            The original lengths of the inputs without padding
         
         Returns
         -------
-        torch.Tensor
+        torch.tensor
             The output tensor of the fine-tuning model.
         """
         outputs = self.model(input_ids=input_ids, attention_mask=attention_mask_minibatch)
-        final_layer = outputs.hidden_states[-1]
-        cls_seq = final_layer[:, 0, :]
-        final = self.fine_tuning_head(cls_seq)
+        batch_embeddings = outputs.hidden_states[-1]
+
+        if self.emb_mode == "cls" and self.cls_present:
+            batch_embeddings = batch_embeddings[:, 0, :]
+        else:
+            length = original_lengths
+            if self.cls_present:
```

Comment:
Are there cases when self.emb_mode != "cls" but self.cls_present is True?

Reply:
Yes, there is CLS present for all geneformer v2 implementations and if we choose a different embedding mode we return the other embeddings excluding the CLS, but it is still present.